src/allocator/memserver_allocator.cpp

```diff
@@ -107,6 +131,28 @@ void Memserver_Allocator::reset_profile() {
     MEMSERVER_PROFILE_START_TIME(NVMM)
 }
 void Memserver_Allocator::dump_profile() { NVMM_PROFILE_DUMP(); }
+
+void Memserver_Allocator::delayed_free_th(uint64_t thread_index)
+{
+    gc_th_struct_t *gc_th_obj = &delayed_free_thread_array[thread_index];
+    while(gc_th_obj->pthread_running == true)
+    {
+		pthread_rwlock_rdlock(&gc_th_obj->rwLock);
+        for (auto itr : *(gc_th_obj->heap_list)) {
```

Comment:
@gautham-bhat-k @soumya-pn  can you verify this - what happens if the current element is removed within this iteration. May be between 148 and 149. In that case gc_th_obj->heap_list might become stale.
Also is there any other window ?